hardware_interface/source/eps.c

```diff
@@ -122,29 +122,29 @@ void prv_instantaneous_telemetry_letoh (eps_instantaneous_telemetry_t *telembuf)
     }
     for (i = 0; i < 4; i++) {
         telembuf->mpptConverterVoltage[i] = csp_letoh16(telembuf->mpptConverterVoltage[i]);
-        telembuf->OutputConverterVoltage[i] = csp_letoh16(telembuf->OutputConverterVoltage[i]);
     }
     for (i = 0; i < 8; i++) {
         telembuf->curSolarPanels[i] = csp_letoh16(telembuf->curSolarPanels[i]);
+        telembuf->OutputConverterVoltage[i] = csp_letoh16(telembuf->OutputConverterVoltage[i]);
     }
-    for (i = 0; i < 10; i++) {
+    for (i = 0; i < 18; i++) {
         telembuf->curOutput[i] = csp_letoh16(telembuf->curOutput[i]);
         telembuf->outputOnDelta[i] = csp_letoh16(telembuf->outputOnDelta[i]);
         telembuf->outputOffDelta[i] = csp_letoh16(telembuf->outputOffDelta[i]);
-        telembuf->outputFaultCnt[i] = csp_letoh32(telembuf->outputFaultCnt[i]);
-    }
-    for (i = 0; i < 12; i++) {
-        telembuf->temp[i] = csp_letoh16(telembuf->temp[i]);
     }
+
     telembuf->vBatt = csp_letoh16(telembuf->vBatt);
     telembuf->curSolar = csp_letoh16(telembuf->curSolar);
     telembuf->curBattIn = csp_letoh16(telembuf->curBattIn);
     telembuf->curBattOut = csp_letoh16(telembuf->curBattOut);
-    telembuf->outputStatus = csp_letoh16(telembuf->outputStatus);
+    telembuf->outputStatus = csp_letoh32(telembuf->outputStatus);
 
-    telembuf->outputFaultStatus = csp_letoh16(telembuf->outputFaultStatus);
+    telembuf->outputFaultStatus = csp_letoh32(telembuf->outputFaultStatus);
     telembuf->wdt_gs_time_left = csp_letoh32(telembuf->wdt_gs_time_left);
     telembuf->wdt_gs_counter = csp_letoh32(telembuf->wdt_gs_counter);
-    telembuf->rstReason = csp_letoh32(telembuf->rstReason);
     telembuf->bootCnt = csp_letoh32(telembuf->bootCnt);
+
+    telembuf->uptimeInS = csp_letoh32(telembuf->uptimeInS);
+    telembuf->PingWdt_toggles = csp_letoh16(telembuf->PingWdt_toggles);
+    telembuf->timestampInS = csp_letoh64(telembuf->timestampInS);

```

Comment:
What I get for the double timestamp is 0.0. I'm gonna test casting because this function return and use uint64, but any other insight? Interestingly, in the ground station '<f8' numpy datatype interprets the data correctly (The issue is not endianness btw).

Reply:
you should be able to take any UNIX time_t timestamp and cast to int32_t or uint32_t. Unless you want milliseconds as the decimal?

Reply:
It seems that it contains fractions of seconds.

Reply:
Well if it's used to do operations multiple times a second then I suppose that's needed. but if it's just used to timestamp data then truncate to uint32_t. the gps and the rtc can't store or provide millis. 
I'd also consider the limitations of our CPU. it's 32-bit and might not be able to process doubles as per this website showing the instruction set only supports single precision float https://developer.arm.com/ip-products/processors/cortex-r/cortex-r5
Processing doubles might be beyond our control

Reply:
The point is it's what EPS gives us. We can ignore those 64 bits, if we don't wanna know the timestamp that the data was recorded from EPS. Or, we can process and use it. I found out the problem was csp functions and I had to write a custom csp_letohd function to convert the endianness correctly. 
What do you mean truncate it to uint32_t? Getting the integer part of a double is a lot more complicated. Our only option if we have limitation for doubles and we wanna get the timestamp anyway would be to get the unix timestamp with another csp request and replace the double one here.

Reply:
Ok so that must be for sampling speeds greater than 1Hz. interesting. Well it's good you found the problem and in that case my point is moot. but I'm pretty sure: 
uint32_t timestamp = (uint32_t)timestampInS; 
would work just fine. Implicit truncation works automatically